solver/src/collections/ref_store.rs

```diff
@@ -562,6 +570,11 @@ impl<K: Ref, V> IterableRefMap<K, V> {
         self.map.insert(k, v)
     }
 
+    pub fn remove(&mut self, k: K) {
```

Comment:
Unused and performance hazard (O(n) complexity) : Remove.

Unless really useful, we should only provide methods where there is no surprises.